### DISCUSSION TITLE
How to capture style attributes as json for randomly generated avatars

### DISCUSSION THREAD

**Discussion by bfxtsales** [dicebear/dicebear#158] [Q&A] [answered]
Very cool project!  I'd like to use it to generate a collection of images and their associated, randomly selected, style attributes as json files on my local drive.  Does this capability exist, and/or has anyone created something along these lines? 

So the goal is to generate 100, say, open-peeps, with randomized styles, and write to disk each "ItemN.png", and an "ItemN.json" which captures the style attributes for each.

So far I've been using the cli interface to generate the randomized images.  Thanks for any help or pointers.

**Comment by FlorianKoerner** [comment, marked as answer]
## Version 5.x
```
npx dicebear openPeeps --count 100 --json
```

https://dicebear.com/how-to-use/cli#passing-options

## Version 4.x
I have put together a small example. You can find it here:
https://github.com/FlorianKoerner/dicebear-158

The trick is that you have to define yourself which style attributes you want to have. You can do this by using the PRNG yourself. Then you create the avatar and put your used options next to it.

My example tries to find out the needed values automatically using the schema.json. But you could also throw away this logic and define yourself what the avatar style receives for options.

<img width="227" alt="Bildschirmfoto 2021-09-19 um 09 51 27" src="https://user-images.githubusercontent.com/647303/133919774-a5057703-b380-421b-bcf0-96a20b02fb41.png">

You can adjust the values in these lines for your needs:
https://github.com/FlorianKoerner/dicebear-158/blob/main/index.ts#L7-L17

And you can run the script as follows:

```bash
# Setup
git clone https://github.com/FlorianKoerner/dicebear-158.git
npm install

# Run
npm start
```

**Comment by bfxtsales** [comment]
Fantastic!  And thank you for the quick response.  Let me work with this for a while.
(I initially posted some git and install problems.  Then I got things working and quickly deleted it, don't know if you saw it :)

**Reply by FlorianKoerner** [reply]
I saw it. 😄  I have updated my answer accordingly.

**Comment by bfxtsales** [comment]
Slight tangent, but what do you use for debugging in typescript?   Also I'm trying to decipher the probabilities section.  I think I get the 0 vs 100 for handling whether or not a given style should be included in options[].  What about the 50% part?  Is that overriding default probabilities?

**Reply by FlorianKoerner** [reply]
I use VSCode. It shows me TypeScript errors immediately. But the script can also be written without TypeScript.

The value 50 is only a fallback if there is no default value. But in the avatar style Open Peeps the value is always set.

The 50 indicates that the probability is 50%. In line 61, this value is then turned into a definite yes or no. So 100 or 0.

**Reply by bfxtsales** [reply]
Ok thanks.  I've been looking for a way to step through typescript natively, haven't found anything.

Also maybe take a look at the defaultProbability section?  I think it's always overwritten with 50?   Might be coming in as a number rather than string?

Thanks again.

**Reply by FlorianKoerner** [reply]
You are right. The type is "number" and not "string". I have corrected this and also configured debugging for VSCode.

<img width="1568" alt="Bildschirmfoto 2021-09-21 um 09 00 23" src="https://user-images.githubusercontent.com/647303/134126171-f6e024b3-c6ec-4733-8a99-dd67a97e501d.png">
